src/box/tuple_format.c

```diff
@@ -1556,6 +1557,9 @@ tuple_format_apply_defaults(struct tuple_format *format, const char **data,
 	tuple_builder_new(&builder, &fiber()->gc);
 	size_t region_svp = region_used(&fiber()->gc);
 	bool is_tuple_changed = false;
+	struct tuple *source =
+		tuple_new(tuple_format_runtime, *data, *data_end);
```

Comment:
I'd consider the following optimization — build a tuple only if there is at least 1 field with default func.

Reply:
Fixed.